doc/reference/ubuntu-image.rst

```diff
@@ -0,0 +1,246 @@
+ubuntu-image syntax and options
+===============================
+
+``ubuntu-image`` is a program for generating bootable disk images. It supports building snap_-based and classical preinstalled Ubuntu images.
+
+
+Snap-based images
+-----------------
+
+Snap-based images are built from a *model assertion*, which is a YAML file describing a particular combination of core, kernel, and gadget snaps, along with other declarations, signed with a digital signature asserting its authenticity.  The assets defined in the model assertion uniquely describe the device for which the image is built.
+
+
+Classical images
+----------------
+
+Classical images are built from a local `gadget tree`_ path.  The `gadget tree`_ is a primed `gadget snap`_ containing a ``gadget.yaml`` file in the ``meta`` directory and all the necessary bootloader gadget bits built. For instance, a `gadget tree`_ can be prepared by fetching a specially tailored `gadget snap`_ source and running ``snapcraft prime`` on it, with the resulting tree being stored in the ``prime/`` directory.
```

Comment:
We still can build the image this way (by providing a pre-built gadget tree) but now the favored way is to set the URL and branch of the gadget in the image definition and let ubuntu-image fetch and build it. So now it feels like "advanced" instructions that may let users think this is the default way to do things. What do you think about moving this at the bottom and replace with:


Classical images are built from an image definition YAML file, defining its content and resulting output format. Key elements of this configuration are:
- the gadget section, defining how to get the gadget required to build boot assets and partition layout.
- the rootfs section, defining parameters to build the content of the rootfs
- the artifacts section, defining the types of artifacts to create, including the actual images, manifest files, etc.